### DISCUSSION TITLE
2.11.3 broke typed redirects

### DISCUSSION THREAD

**Discussion by mabdullahsari** [livewire/livewire#5568] [7. Bugs (archived)] [answered]
'ello 👋

[This change](https://github.com/livewire/livewire/blob/9da51085a7b02a8512a9d4c525d5ea96e8f52aa0/src/Controllers/HttpConnectionHandler.php#L78) from noop to 204 responses causes redirects to be ignored that are produced as a result of a contract.

Basically, this component no longer works:

```php
final class EmailVerificationCheck extends Component
{
    use WithUser;

    public function check(): ?\Laravel\Fortify\Contracts\VerifyEmailResponse
    {
        if ($this->user->hasVerifiedEmail()) {
            return app(VerifyEmailResponse::class);
        }

        $this->skipRender();

        return null;
    }

    public function render(Factory $view): View
    {
        return $view->make('id::verify-email.email-verification-check');
    }
}
```

`VerifyEmailResponse` is a contract defined by Laravel Fortify, extending `Responsable` which produces an `\Illuminate\Http\RedirectResponse`.

**Comment by joshhanley** [comment]
@mabdullahsari thanks for reporting! 

Any chance you could create a small example repo which reproduces the issue with as little code as possible so we can investigate it?

Thanks!

**Reply by mabdullahsari** [reply]
@joshhanley Sure, here you go:

https://github.com/dive-be/livewire-5568-repro

**Reply by joshhanley** [reply]
@mabdullahsari thanks! Much appreciated. I will have a look and see if I can work out what is going on.

**Comment by joshhanley** [comment]
@mabdullahsari thanks so much for the repo! Made it much easier for me to track the issue down.

I've submitted a PR #5574 with a fix.

**Reply by mabdullahsari** [reply]
Hey man, no problem! Thank you for taking swift action. 🙏

**Reply by joshhanley** [reply]
No worries! 🙂

**Comment by joshhanley** [comment, marked as answer]
Fix for this has been merged in PR #5574 and tagged in v2.12.1.
